java/src/test/java/org/extism/sdk/PluginTests.java

```diff
@@ -104,4 +110,53 @@ public void shouldAllowInvokeFunctionFromFileWasmSourceMultipleTimesByReusingCon
         }
     }
 
+    @Test
+    public void shouldAllowInvokeHostFunctionFromPDK() {
+        var parametersTypes = new LibExtism.ExtismValType[]{LibExtism.ExtismValType.I64};
+        var resultsTypes = new LibExtism.ExtismValType[]{LibExtism.ExtismValType.I64};
+
+        class MyUserData extends HostUserData {
+            private String data1;
+            private int data2;
+
+            public MyUserData(String data1, int data2) {
+                super();
+                this.data1 = data1;
+                this.data2 = data2;
+            }
+        }
+
+        ExtismFunction helloWorldFunction = (ExtismFunction<MyUserData>) (plugin, params, returns, data) -> {
+            System.out.println("Hello from Java Host Function!");
+            System.out.println(String.format("Input string received from plugin, %s", plugin.inputString(params[0])));
+
+            int offs = plugin.alloc(4);
+            Pointer mem = plugin.memory();
+            mem.write(offs, "test".getBytes(), 0, 4);
+            returns[0].v.i64 = offs;
+
+            System.out.println(String.format("Host user data, %s, %d", data.data1, data.data2));
+        };
+
+        HostFunction hello_world = new HostFunction<>(
+                "hello_world",
+                parametersTypes,
+                resultsTypes,
+                helloWorldFunction,
+                new MyUserData("test", 2)
+        );
+
+        HostFunction[] functions = {hello_world};
```

Comment:
I'm not sure the best way to do this in java as my java knowledge is quite outdated, but is it possible to make the generic type optional? So you could pass none or null to the userdata param? Something like this:

```

        HostFunction null_user_data = new HostFunction(
                "hello_world",
                parametersTypes,
                resultsTypes,
                helloWorldFunction,
                null
        );
```

There are definitely cases where a host function doesn't need a user data so it might be cumbersome to have someone make a type. Or maybe we provide some kind of none type for them if there is no generic way to handle it.